### ISSUE TITLE
Build issue when install Gulp-cordova-build

### ISSUE THREAD

**Issue by mhartington** [polotek/libxmljs#303]
https://github.com/MichaelTaylor3D/gulp-cordova-bump

So I got this error when I was trying to install a package that requires libxmljs.
Figured I open an issue here and on the packages repo.

```
2015-04-09 18:04:03.678 xcodebuild[40174:696370] [MT] PluginLoading: Required plug-in compatibility UUID 9F75337B-21B4-4ADC-B558-F9CADF7073A7 for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Alcatraz.xcplugin' not present in DVTPlugInCompatibilityUUIDs
  CC(target) Release/obj.target/libxml/vendor/libxml/buf.o
  CC(target) Release/obj.target/libxml/vendor/libxml/catalog.o
  CC(target) Release/obj.target/libxml/vendor/libxml/chvalid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/dict.o
  CC(target) Release/obj.target/libxml/vendor/libxml/encoding.o
  CC(target) Release/obj.target/libxml/vendor/libxml/entities.o
  CC(target) Release/obj.target/libxml/vendor/libxml/error.o
  CC(target) Release/obj.target/libxml/vendor/libxml/globals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/hash.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLparser.o
  CC(target) Release/obj.target/libxml/vendor/libxml/HTMLtree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/legacy.o
  CC(target) Release/obj.target/libxml/vendor/libxml/list.o
  CC(target) Release/obj.target/libxml/vendor/libxml/parser.o
../vendor/libxml/parser.c:13492:43: warning: passing 'const xmlChar *' (aka 'const unsigned char *') to parameter
      of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        hdlr = xmlFindCharEncodingHandler(doc->encoding);
                                          ^~~~~~~~~~~~~
../vendor/libxml/include/libxml/encoding.h:175:42: note: passing argument to parameter 'name' here
        xmlFindCharEncodingHandler      (const char *name);
                                                     ^
1 warning generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/parserInternals.o
  CC(target) Release/obj.target/libxml/vendor/libxml/pattern.o
  CC(target) Release/obj.target/libxml/vendor/libxml/relaxng.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX2.o
  CC(target) Release/obj.target/libxml/vendor/libxml/SAX.o
  CC(target) Release/obj.target/libxml/vendor/libxml/tree.o
  CC(target) Release/obj.target/libxml/vendor/libxml/threads.o
  CC(target) Release/obj.target/libxml/vendor/libxml/uri.o
  CC(target) Release/obj.target/libxml/vendor/libxml/valid.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xinclude.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xlink.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlIO.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmemory.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlmodule.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlreader.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlregexp.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlsave.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemas.o
../vendor/libxml/xmlschemas.c:24203:10: warning: explicitly assigning value of variable of type 'xmlSchemaValType'
      to itself [-Wself-assign]
        valType = valType;
        ~~~~~~~ ^ ~~~~~~~
1 warning generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlschemastypes.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlstring.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlunicode.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xmlwriter.o
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
../vendor/libxml/xpath.c:12402:13: warning: comparison of constant 'XML_NAMESPACE_DECL' (18) with expression of
      type 'xmlXPathTypeVal' is always false [-Wtautological-constant-out-of-range-compare]
                        if (type == XML_NAMESPACE_DECL)
                            ~~~~ ^  ~~~~~~~~~~~~~~~~~~
In file included from ../vendor/libxml/xpath.c:185:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity' [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
3 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:63:
../node_modules/nan/nan_new.h:207:43: error: inline declaration of 'NanNew' follows non-inline definition
template <typename T> inline v8::Local<T> NanNew(v8::Handle<T> h);
                                          ^
../node_modules/nan/nan_implementation_pre_12_inl.h:210:1: note: previous definition is here
NanNew(v8::Handle<T> h) {
^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:63:
../node_modules/nan/nan_new.h:208:43: error: inline declaration of 'NanNew' follows non-inline definition
template <typename T> inline v8::Local<T> NanNew(v8::Persistent<T> const& p);
                                          ^
../node_modules/nan/nan_implementation_pre_12_inl.h:216:1: note: previous definition is here
NanNew(v8::Persistent<T> const& p) {
^
2 errors generated.
make: *** [Release/obj.target/xmljs/src/libxmljs.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Darwin 14.3.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/mhartington/gulp-cordova-bump/node_modules/gulp-xml-editor/node_modules/libxmljs
gyp ERR! node -v v0.10.38
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok
npm ERR! Darwin 14.3.0
npm ERR! argv "node" "/usr/local/bin/npm" "install"
npm ERR! node v0.10.38
npm ERR! npm  v2.7.5
npm ERR! code ELIFECYCLE

npm ERR! libxmljs@0.13.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the libxmljs@0.13.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the libxmljs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls libxmljs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/mhartington/gulp-cordova-bump/npm-debug.log
```

The full npm output is available here
https://gist.github.com/mhartington/e1fa2c107c53d499f0bd

OSX 10.10.3
Node 0.10.38

What other information do you need?

**Comment by defunctzombie** [comment]
Those look like NAN errors to me. ping @rvagg

**Comment by lmarkus** [comment]
Seconded...
In my case `libxmljs` is a dependency on another library. Installation of `0.13.0` fails with the above, but `0.12.0`  is ok.

OSX Yosemite
NPM 2.7.5
Node 0.10.38
Python 2.7.9

**Comment by aredridel** [comment]
Which GCC/clang?

**Comment by lmarkus** [comment]
```bash
$ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.1.0 (clang-602.0.49) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.1.0
Thread model: posix
```

**Comment by aredridel** [comment]
Can confirm:

```
npm install node-bin@0.10.38
npm install libxmljs
```

Fails at 

```
  CC(target) Release/obj.target/libxml/vendor/libxml/xpath.o
../vendor/libxml/xpath.c:12402:13: warning: comparison of constant 'XML_NAMESPACE_DECL' (18) with
      expression of type 'xmlXPathTypeVal' is always false
      [-Wtautological-constant-out-of-range-compare]
                        if (type == XML_NAMESPACE_DECL)
                            ~~~~ ^  ~~~~~~~~~~~~~~~~~~
In file included from ../vendor/libxml/xpath.c:185:
../vendor/libxml/trionan.c:194:1: warning: unused function 'trio_is_special_quantity'
      [-Wunused-function]
trio_is_special_quantity
^
../vendor/libxml/trionan.c:218:1: warning: unused function 'trio_is_negative' [-Wunused-function]
trio_is_negative
^
3 warnings generated.
  CC(target) Release/obj.target/libxml/vendor/libxml/xpointer.o
  LIBTOOL-STATIC Release/xml.a
  CXX(target) Release/obj.target/xmljs/src/libxmljs.o
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:63:
../node_modules/nan/nan_new.h:207:43: error: inline declaration of 'NanNew' follows non-inline
      definition
template <typename T> inline v8::Local<T> NanNew(v8::Handle<T> h);
                                          ^
../node_modules/nan/nan_implementation_pre_12_inl.h:210:1: note: previous definition is here
NanNew(v8::Handle<T> h) {
^
In file included from ../src/libxmljs.cc:7:
In file included from ../src/libxmljs.h:7:
In file included from ../node_modules/nan/nan.h:63:
../node_modules/nan/nan_new.h:208:43: error: inline declaration of 'NanNew' follows non-inline
      definition
template <typename T> inline v8::Local<T> NanNew(v8::Persistent<T> const& p);
                                          ^
../node_modules/nan/nan_implementation_pre_12_inl.h:216:1: note: previous definition is here
NanNew(v8::Persistent<T> const& p) {
^
2 errors generated.
make: *** [Release/obj.target/xmljs/src/libxmljs.o] Error 1
```

**Comment by aredridel** [comment]
And master works, with nan 1.7.0.

**Comment by lmarkus** [comment]
@polotek @defunctzombie Looks like you just need to publish out a new version to npm.
The code here in master works fine.

**Comment by bluelnkd** [comment]
Getting this here as well. Temporary work around is locking libxmljs as mentioned to 0.12.0.

**Comment by fa137** [comment]
^ can confirm, locking it is the way to handle this error temporarily.

**Comment by laustdeleuran** [comment]
+1

**Comment by defunctzombie** [comment]
new version was published as 0.14.0

**Comment by olekus2** [comment]
I had same problem. Just try command "npm install libxmljs" inside your project with package.json. And then it will install gulp and another packages. after this command try to "gulp" in place where is your gulpfile.js is.
